### PR TITLE
[fli-iam#1671] Optimize preclinical subjects loading

### PR DESCRIPTION
**Pull request by youennmerel**

### CODE REVIEW

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java [hunk 1]
```diff
@@ -92,11 +93,22 @@ public ResponseEntity<SubjectDTO> findSubjectById(
 	}
 
 	@Override
-	public ResponseEntity<List<SubjectDTO>> findSubjects() {
-		final List<Subject> subjects = subjectService.findAll();
+	public ResponseEntity<List<SubjectDTO>> findSubjects(boolean preclinical, boolean clinical) {
```
Comment (julien-louis) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java:96]: Is this necessary to have two booleans when a subject is either clinical or preclinical ?

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApi.java [hunk 2]
```diff
@@ -68,7 +70,12 @@ ResponseEntity<Void> deleteSubject(
 	@GetMapping(value = "", produces = { "application/json" })
 	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT', 'USER')")
 	@PostAuthorize("hasRole('ADMIN') or @studySecurityService.filterSubjectDTOsHasRightInOneStudy(returnObject.getBody(), 'CAN_SEE_ALL')")
-	ResponseEntity<List<SubjectDTO>> findSubjects();
+	ResponseEntity<List<SubjectDTO>> findSubjects(
```
Comment (julien-louis) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApi.java:73]: same : Is this necessary to have two booleans when a subject is either clinical or preclinical ?

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java (hunk 1 above)
Comment (youennmerel) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java:96]: Hi @julien-louis,

I thought of 3 possible uses cases :
- we want all subject (default values : clinical = true & preclinical = true)
- we want only clinical (clinical = true & preclinical = false)
- we want only preclinical (clinical = false & preclinical = true)

However, for now, API is only used for getting all subjects, or just preclinical one.

Do you think of a more elegant solution ?

**Comment by youennmerel** [comment]
Loading time is now ~ 1-2 seconds in qualif.

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java (hunk 1 above)
Comment (julien-louis) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/controler/SubjectApiController.java:96]: Ok I see now, seems fine, thank you

**Review by julien-louis — APPROVED** [review]

**Pull request merged**